lib/mkEnv.nix

```diff
@@ -91,11 +93,18 @@ let
       builder = "/bin/sh";
       args = [
         "-c"
-        "echo ${env}; ${coreutils}/bin/mkdir $out; ${coreutils}/bin/cp ${
+        "echo ${env}; ${coreutils}/bin/mkdir -p $out/nix-support; ${coreutils}/bin/cp ${
           if manifestPath == null
           then manifestFile
           else manifestPath
-        } $out/manifest.json"
+        } $out/manifest.json; echo ${
+          # TODO: get the primary output from the [ordered] eval.outputs
+          lib.concatMapStringsSep " " (x: x.meta.publishData.eval.outputs.out) (
+            builtins.filter (
+	      x: lib.hasAttrByPath [ "meta" "publishData" "eval" "outputs" "out" ] x
+	    ) args.packages
+          )
+        } > $out/nix-support/propagated-user-env-packages"
```

Comment:
Not a fan of these rather complex nix expressions inline in a bash script that is a (nix) list item of arguments passed to sh..

just a lot of indirections... maybe bring them in though a let binding rather than inline.

also were mixing-in tabs here

Reply:
Tabs were definitely a mistake - thanks for catching. If you have a suggested re-working of the above would be happy to accept those, otherwise if you're OK with the above we can move forward with that (it does work).

Reply:
How about just separating things:

```    
    # TODO: get the primary output from the [ordered] eval.outputs
    propagatedPackages = lib.concatMapStringsSep " " (x: 
      x.meta.publishData.eval.outputs.out) (
        builtins.filter (
          x: lib.hasAttrByPath [ "meta" "publishData" "eval" "outputs" "out" ] x
        ) args.packages
      );
    buildScript = ''
      echo ${env};
      ${coreutils}/bin/mkdir -p $out/nix-support;
      ${coreutils}/bin/cp ${
        if manifestPath == null
          then manifestFile
          else manifestPath
      } $out/manifest.json; 
      echo ${propagatedPackages} > $out/nix-support/propagated-user-env-packages;
    '';
    manifest = derivation {
      name = "profile";
      inherit system;
      builder = "/bin/sh";
      args = [
        "-c"
        buildScript
      ];
    };
```